asr-python-advanced-finetune-am-conformer-ctc-tao-finetuning-librispeech-nigerian.ipynb

```diff
@@ -0,0 +1,971 @@
+    "change_vocabulary: false\n",
+    "\n",
+    "tokenizer:\n",
+    "  dir: ???\n",
+    "  type: \"bpe\"  # Can be either bpe or wpe\n",
+    "\n",
+    "# Fine-tuning settings: training dataset\n",
+    "finetuning_ds:\n",
+    "  manifest_filepath: ???\n",
+    "  batch_size: 4\n",
+    "  trim_silence: true\n",
+    "  shuffle: true\n",
+    "  is_tarred: false\n",
+    "  tarred_audio_filepaths: null\n",
+    "\n",
+    "# Fine-tuning settings: validation dataset\n",
+    "validation_ds:\n",
+    "  manifest_filepath: ???\n",
+    "  batch_size: 4\n",
+    "  shuffle: false\n",
+    "\n",
+    "# Fine-tuning settings: optimizer\n",
+    "optim:\n",
+    "  name: novograd\n",
+    "  lr: 0.001\n",
+    "EOF"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Finetune Conformer-CTC"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!tao speech_to_text_conformer finetune \\\n",
+    "     -e $SPECS_DIR/conformer/finetune_en_ng.yaml \\\n",
+    "     -g 1 \\\n",
+    "     -k $KEY \\\n",
+    "     -m $RESULTS_DIR/conformer/train/checkpoints/trained-model.tlt \\\n",
+    "     -r $RESULTS_DIR/conformer/finetune \\\n",
+    "     finetuning_ds.manifest_filepath=$DATA_DIR/en_ng_ft_manifest.json \\\n",
+    "     validation_ds.manifest_filepath=$DATA_DIR/en_ng_val_manifest.json \\\n",
+    "     trainer.max_epochs=1 \\\n",
+    "     finetuning_ds.num_workers=20 \\\n",
+    "     validation_ds.num_workers=20 \\\n",
+    "     trainer.gpus=1 \\\n",
+    "     tokenizer.dir=$DATA_DIR/LibriSpeech/train-clean-100/tokenizer_spe_unigram_v1024"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### ASR model export"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "With TAO, you can also export your model in a format that can deployed using NVIDIA Riva; a highly performant application framework for multi-modal conversational AI services using GPUs. The same command for exporting to ONNX can be used here. The only small variation is the configuration for `export_format` in the spec file."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Export to Riva"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!tao speech_to_text_conformer export \\\n",
+    "     -e $SPECS_DIR/conformer/export.yaml \\\n",
+    "     -g 1 \\\n",
+    "     -k $KEY \\\n",
+    "     -m $RESULTS_DIR/conformer/train/checkpoints/trained-model.tlt \\\n",
+    "     -r $RESULTS_DIR/conformer/riva \\\n",
+    "     export_format=RIVA \\\n",
+    "     export_to=asr-model.riva"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Export to ONNX (Note: Export to ONNX is not needed for Riva)"
```

Comment:
Ideally, we move the Note out of the heading title.

Reply:
Done. I'll have to make that change in all of the fine-tuning notebooks, though.

Reply:
All right, I've made the change in all the relevant notebooks. I'm going to mark the conversation as resolved.